spec/lib/manageiq/active_record_connector_spec.rb

```diff
@@ -0,0 +1,152 @@
+    it "returns true when a connection is already established" do
+      expect(described_class.connection_exists?).to be true
+    end
+
+    it "returns false when a connection does already not exist" do
+      cmd = "print #{described_class}.connection_exists?"
+      expect(without_rails(cmd)).to eq("false")
+    end
+  end
+
+  describe ".establish_connection_if_needed" do
+    # Save off existing setup
+    let!(:config)     { ActiveRecord::Base.connection_config }
+    let!(:connection) { ActiveRecord::Base.connection }
+    let!(:logger)     { ActiveRecord::Base.logger }
+
+    it "doesn't create a new connection/logger if it doesn't need to" do
+      described_class.establish_connection_if_needed(config, logger.filename)
+
+      expect(ActiveRecord::Base.connection).to eq(connection)
+      expect(ActiveRecord::Base.logger).to     eq(logger)
+    end
+
+    it "creates a new connection as needed" do
+      cmd = [
+        "#{described_class}.establish_connection_if_needed(#{config.inspect.gsub(/(")/, "\\\\\\1")}, '#{logger.filename}')",
+        "puts #{described_class}.connection_exists?",
+        # We aren't using vmdb-logger in the without_rails section of the code,
+        # so the `filename` and `logdev` methods aren't available like they
+        # are in VmdbLogger, which is what our logger is for rails.  Have to
+        # do it this way in the script because of that.
+        "puts ActiveRecord::Base.logger.instance_variable_get(:@logdev).filename"
+      ].join("; ")
+
+      expected_output = "true\n#{logger.filename}"
+      expect(without_rails(cmd)).to eq(expected_output)
+    end
+
+    it "uses a default logger if a log path isn't supplied" do
+      cmd = [
+        "#{described_class}.establish_connection_if_needed(#{config.inspect.gsub(/(")/, "\\\\\\1")})",
+        "puts #{described_class}.connection_exists?",
+        "puts ActiveRecord::Base.logger.instance_variable_get(:@logdev).filename"
+      ].join("; ")
+
+      expected_output = "true\n#{ManageIQ.root.join "log", "test.log"}"
+      expect(without_rails(cmd)).to eq(expected_output)
+    end
+
+    it "reuses the existing logger if that is already set" do
+      cmd = [
+        "ActiveRecord::Base.logger = Logger.new '#{ManageIQ.root.join "log", "foobar.log"}'",
+        "#{described_class}.establish_connection_if_needed(#{config.inspect.gsub(/(")/, "\\\\\\1")})",
+        "puts #{described_class}.connection_exists?",
+        "puts ActiveRecord::Base.logger.instance_variable_get(:@logdev).filename"
+      ].join("; ")
+
+      expected_output = "true\n#{ManageIQ.root.join "log", "foobar.log"}"
+      expect(without_rails(cmd)).to eq(expected_output)
+    end
+
+    it "favors DATABASE_URL over database.yml" do
+      env = {"DATABASE_URL" => "postgres://root@localhost:12345/test"}
+      cmd = [
+        # This next line is a HACK.  Remove before merging
+        "module Vmdb; class Settings; def self.decrypt_passwords!(arg); arg; end; end; end;",
```

Comment:
This line is only to allow the line in `database_configuration_patch.rb` to function (and spit back what it was given) without needing to involve `vmdb/logging` and loading all of `vmdb/settings` to make it function.  Many changes are required to make that work, so I am looking at a different code change to avoid doing all that at the moment.

This line shouldn't really make it into master.